python/lsst/ts/MTAOS/MtaosCsc.py

```diff
@@ -65,7 +72,10 @@ def __init__(self, config_dir=None, logToFile=False, simulation_mode=0):
                          simulation_mode=int(simulation_mode))
 
         # Logger attribute comes from the upstream Controller class
-        self.log = self._addLogWithFileHandlerIfDebug(outputLogFile=logToFile)
+        if (debug_level is None):
```

Comment:
Even better, I don't see reason for this. You can use logger.getLogger and customize this if realy needed. I will vote to remove this hack (unless there is truly a reason for it, which I miss so far).

Reply:
1. The `log` attribute of `MtaosCSC` class is inherited from the upstream `ts_salobj`, which will put the log message into the EFD. In the component tests happened at Dec., 2019, I realized to output the log message in the local machine is easier to debug. Therefore, I decided to add the support of log file.

2. For the return of `log` object, it is intentional. If you read the `Utility. addRotFileHandler()`, you will realize from the initial beginning, the object of `log` is always the same. This implies the return of `log` is not necessary actually. However, the book of "Clean Code" or "Code Complete 2" (I forgot which one) suggested this method for the maintenance reason. This is to let the future's developer/ maintainer know a object's behavior is changed (for example, the object's attribute value is updated already after the function call). I think this way is very helpful in the point of maintaince.

3. When we used the CSC, we will go through the command link task (`bin/run_mtaos.py`). Therefore, the default value of `debug_level` is not decided at `MtaosCsc.__init__()`. Actually, I did the same thing in the initial beginning and I realized I could not do this. The default values are decided at `add_arguments()` and `add_kwargs_from_args()` actually. Since I set the data type of `--debugLevel` to be `int`, it will return `None` if the user did not use this option. You need to know the implementation of CSC based on `ts_salobj` is better to follow its own logic for the future's maintenance.

Thanks!

Reply:
BTW, I just remembered another reason to expose the return of `self.log` even it is not needed. This is based on the book of Clean Code that the developer/ maintainer might be used to read the construction function (e.g. `__init__()`) to expect the available internal attribute to use. However, this `self.log` in inherited from the upstream class actually. Therefore, I decided to expose it to let the future's maintainer knows he/she has this `log` attribute to use to put the message into the EFD or a file.